lib/glimesh/chat/policy.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Glimesh.Chat.Policy do
+  @moduledoc """
+  Glimesh Chat Policy
+
+
+  """
+
+  @behaviour Bodyguard.Policy
+
+  import GlimeshWeb.Gettext
+
+  alias Glimesh.Accounts.User
+  alias Glimesh.Chat
+  alias Glimesh.Streams.Channel
+
+  def authorize(:create_chat_message, %User{is_admin: true}, _channel), do: true
+
+  def authorize(:create_chat_message, %User{id: user_id}, %Channel{user_id: channel_user_id})
+      when user_id == channel_user_id,
+      do: true
+
+  def authorize(:create_chat_message, %User{} = user, %Channel{} = channel) do
+    cond do
+      # Specific Channel Ban
+      expiry = Chat.is_banned_until(channel, user) ->
+        if expiry == :infinity do
+          {:error, gettext("You are permanently banned from this channel.")}
+        else
+          seconds = NaiveDateTime.diff(expiry, NaiveDateTime.utc_now(), :second)
+
+          {:error,
+           gettext("You are banned from this channel for %{minutes} more minutes.",
+             minutes: round(Float.ceil(seconds / 60))
+           )}
+        end
+
+      # Global Account Ban
+      Glimesh.Accounts.is_user_banned?(user) ->
+        {:error, gettext("You are banned from Glimesh.")}
+
+      true ->
+        true
+    end
+  end
+
+  # Admins
+  def authorize(:ban, %User{is_admin: true}, _channel), do: true
+  def authorize(:unban, %User{is_admin: true}, _channel), do: true
+  def authorize(:short_timeout, %User{is_admin: true}, _channel), do: true
+  def authorize(:long_timeout, %User{is_admin: true}, _channel), do: true
+
+  # Channel Owners
+  def authorize(:ban, %User{id: user_id}, %Channel{user_id: channel_user_id})
+      when user_id == channel_user_id,
+      do: true
```

Comment:
Just a curiosity, is there any way of checking that the user is the channel owner without doing a when? Either way this works without issues it seems, just curious if there's a better way for quick readability.

Reply:
You could do it in the `do` block as well, but it would return false instead if skipping the function, which means the moderators code below would never trigger.